### ISSUE TITLE
Attribute declarations should support quoted text

### ISSUE THREAD

**Issue by gdamore** [asciidoctor/asciidoctor#3708]
Suppose I would like to have a space at the end of my attribute value like this:

```
:my-prefix: 'Evidence '
```

This should, I think, be treated as a quoted string, and when expanded in this:

```
{my-prefix}1. Conditional spaces
```

Wind up getting expanded as

```
Evidence 1. Conditional spaces
```

Instead, it gets expanded as 

```
'Evidence '1. Conditional spaces
```

An equivalent thing happens for double quotes.  So it appears that there is no way to end an attribute with a space in this style of declaration.  I *suspect* that this is simply an oversight.  It seems like it would be better to allow a quoted string here.

If quoted strings were allowed here, then also newlines could be embedded which would also be nice (I do realize we have special rules for backslash escaping newlines.)

**Comment by gdamore** [comment]
I also, since we're on the topic, find that the rule that backslashes get emitted as is, except when followed by a newline, to be particularly strange.  Admittedly I'm coming at this from a programmer's perspective, but it seems like backslashes should just always escape the next character, with double backslash used to represent a single one in the output.

**Comment by Mogztter** [comment]
Supporting quoted text might have other benefits but for your particular use case I think you should use the attribute `{nbsp}`.

In my opinion, it's also better than a space in this context because you don't want a word break between "Evidence" and "1".

```adoc
:my-prefix: Evidence{nbsp}

{my-prefix}1. Conditional spaces
```

**Comment by gdamore** [comment]
I agree in this particular case that's true.  I was trying to come up with a use case for the space at the end that wasn't too contrived.  :/

**Comment by mojavelinux** [comment]
It's already possible to apply substitutions to the value of the attribute entry. The value must be enclosed in the pass macro with the substitutions specified on the macro.

See https://asciidoctor.org/docs/user-manual/#altering-attribute-entry-substitutions

Examples are given in the documentation. Single and double quotes have no special meaning in the value of an attribute entry. Adding that now would be a major breaking change to the language, so could only be introduced by the versioned specification.

**Comment by gdamore** [comment]
That's fine.  Hopefully these ideas can be presented when the time comes to write the specification.

I realize the place we're in now evolved organically.  I only hope the project can resist the temptation to add more syntactic sugar to the language until some kind of work towards a formal specification can be made.

**Comment by mojavelinux** [comment]
> I only hope the project can resist the temptation to add more syntactic sugar to the language until some kind of work towards a formal specification can be made.

That's exactly what I'm saying, so we're in agreement there.
